tests/rptest/tests/partition_reassignments_test.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2023 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+import random
+
+from ducktape.utils.util import wait_until
+from rptest.services.cluster import cluster
+from rptest.tests.redpanda_test import RedpandaTest
+
+from rptest.clients.types import TopicSpec
+from rptest.clients.kafka_cli_tools import KafkaCliTools
+
+
+class PartitionReassignmentsTest(RedpandaTest):
+    PARTITION_COUNT = 3
+    topics = [
+        TopicSpec(partition_count=PARTITION_COUNT),
+        TopicSpec(partition_count=PARTITION_COUNT),
+    ]
+
+    def __init__(self, test_context):
+        super(PartitionReassignmentsTest,
+              self).__init__(test_context=test_context, num_brokers=4)
+
+    def get_missing_node_idx(self, lhs: list[int], rhs: list[int]):
+        missing_nodes = set(lhs).difference(set(rhs))
+        assert len(missing_nodes) == 1, "Expected one missing node"
+        return missing_nodes.pop()
+
+    @cluster(num_nodes=4)
+    def test_reassignments(self):
+        initial_assignments = self.client().describe_topics()
+        self.logger.debug(f"Initial assignments: {initial_assignments}")
+
+        all_node_idx = [
+            self.redpanda.node_id(node) for node in self.redpanda.nodes
+        ]
+        self.logger.debug(f"All node idx: { all_node_idx}")
+
+        reassignments_json = {"version": 1, "partitions": []}
```

Comment:
`kafka-reassign-partitions.sh` does include an option to auto-generate this json file; however, we have less control over the reassignment results. The generate option only allows for "adding node ids" which means there is a chance that some partitions may not have reassignments.

Therefore, we can force a reassignment on all partitions by manually creating this JSON object. IMO this is a better test than using the generate flag.

Are there any conflicting opinions?